src/util/logger.ts

```diff
@@ -58,20 +89,14 @@ class Logger {
     this.crawlState = crawlState;
   }
 
-  // TODO: Fix this the next time the file is edited.
-
   logAsJSON(
     message: string,
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    data: Record<string, string> | Error | any,
+    dataUnknown: unknown,
     context: string,
     logLevel = "info",
   ) {
-    if (data instanceof Error) {
-      data = errJSON(data);
-    } else if (typeof data !== "object") {
-      data = { message: data.toString() };
-    }
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const data: Record<string, any> = errJSON(dataUnknown);
```

Comment:
Wondering if we should rename `errJSON()` now that it's being used to format messages for all log levels? Otherwise this is a nice solution for getting a consistent type from an unknown (and I got to learn about `unknown`, so thanks :) )

Reply:
This was referring to formatting the exception rather than the log level specifically

Reply:
The name came from it checking `if (e instanceof Error)` and converting that to json.. not sure if there's a better name now - toJSON() seems overly generic

Reply:
Converted to `formatErr`